airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/GcsDestination.java

```diff
@@ -31,13 +36,38 @@ public static void main(final String[] args) throws Exception {
   @Override
   public AirbyteConnectionStatus check(final JsonNode config) {
     try {
-      final GcsDestinationConfig destinationConfig = GcsDestinationConfig.getGcsDestinationConfig(config);
+      final String fileName = "test";
+      final String testContent = "check-content";
+      final GcsDestinationConfig destinationConfig = GcsDestinationConfig
+          .getGcsDestinationConfig(config);
       final AmazonS3 s3Client = GcsS3Helper.getGcsS3Client(destinationConfig);
-      s3Client.putObject(destinationConfig.getBucketName(), "test", "check-content");
-      s3Client.deleteObject(destinationConfig.getBucketName(), "test");
+      LOGGER.info("Started testing if all required credentials assigned to user");
+      s3Client.putObject(destinationConfig.getBucketName(), fileName, testContent);
+      s3Client.deleteObject(destinationConfig.getBucketName(), fileName);
+      LOGGER.info("Finished checking for normal upload, started checking multipart uploading...");
+      TransferManager tm = TransferManagerBuilder.standard()
+          .withS3Client(s3Client)
+          .build();
+
+      // Test Multipart Upload permissions
+      // TransferManager processes all transfers asynchronously,
+      // so this call returns immediately.
+      final File tmpFile = File.createTempFile(fileName, ".tmp");
+      final FileWriter writer = new FileWriter(tmpFile);
+      writer.write(testContent);
+      writer.close();
```

Comment:
Could we use try-with-resources here, instead of `.close()`method?

Reply:
Good catch, updated, thanks